packages/medusa/src/api/store/products/route.ts

```diff
@@ -80,7 +80,7 @@ async function getProductsWithIndexEngine(
   await wrapProductsWithTaxPrices(req, products)
   res.json({
     products,
-    count: metadata!.count,
+    count: metadata!.estimate_count,
```

Comment:
thought: Would it make sense to return the estimated count more explicitly here when the index engine is used?
```
{ 
  products: [ ... ], 
  estimated_count: 1000,
  count: 1000, // ensures compatibility with existing implementations
  offset: 0, 
  limit: 50 
}
```

This would make the behavior more obvious to the end-user. I fear using `count` for the estimate could be confusing in the presence of incorrect results.

For example, the case we had with the estimated count being 56 and the actual number of rows 1.

Reply:
Yeah, that's better.
I added this property as optional to the paginated response.